docs/architecture/adr-007-specialization-groups.md

```diff
@@ -0,0 +1,161 @@
+# ADR 007: Specialization Groups
+
+## Changelog
+
+- 2019 Jul 31 : Initial Conception
+
+## Context
+
+This idea was first conceived of in order to fulfill the use case of the
+creation of a decentralized Computer Emergency Response Team (CERT), Whose
+members would be elected by the Cosmos Community and would fulfill the role of
+coordinating the Cosmos community under an emergency situations. This thinking
+can be further abstracted into the conception of "blockchain specialization
+groups". 
+
+The creation of these groups are the beginning of specialization capabilities
+within a wider blockchain community which could be used to enable a certain
+level of delegated responsibilities. Examples of specialization which could be
+beneficial to a blockchain community include: code auditing, emergency response,
+code development etc. This type of community organization paves the way for
+individual stakeholders to delegate votes by issue type, if in the future
+governance proposals include a field for issue type. 
+
+
+## Decision
+
+A specialization group can be broadly broken down into
+the following functions: 
+ - Membership Admittance
+   - via governance
+   - via special appointment
+ - Membership Acceptance
+ - Membership Revocation
+   - (probably) Without Penalty 
+     - member steps down (self-Revocation)
+     - replaced by new member from governance
+   - (probably) With Penalty 
+     - due to breach of soft-agreement (determined through governance)
+     - due to breach of hard-agreement (determined by code) 
+ - Execution of Duties
+   - Special transactions which only execute for members of a specialization group
+ - Compensation
+   - Group compensation (further distribution decided by the specialization group) 
+   - Individual compensation for all constituents of a group from the
+     greater community
+
+Election of the members of a specialization group can happen in a wide variety
+of ways and be subject to an arbitrary number of associated rules. For its
+initial implementation as a part of this ADR we recommend that a general
+election abstraction (`Electionator`) is provided as well as a basic
+implementation of that abstraction which allows for a continuous election of
+members of a specialization group. 
+
+``` golang
+// The Electionator abstraction covers the concept space for 
+// a wide variety of election kinds.  
+type Electionator interface {
+    
+    // is the election object accepting votes.
+    Active() bool 
+
+    // functionality to execute for when a vote is cast in this election, here
+    // the vote field is anticipated to be marshalled into a vote type used 
+    // by an election. 
+    Vote(addr sdk.AccAddress, vote []byte) 
```

Comment:
and the proposal id?

Reply:
I don't think that the proposal id is required here, because this is a vote _on_ this Electionator, each electionator is not anticipated to have multiple proposals within it.

Reply:
👍 that needs to be clarified in the comments too

Reply:
I added more discussion to this effect here